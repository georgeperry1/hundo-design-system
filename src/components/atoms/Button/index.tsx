import React, { ReactChild } from 'react';

import BaseButton from './styled';

export type ButtonProps = {
  isFullWidth?: boolean;
  htmlType?: React.ButtonHTMLAttributes<HTMLButtonElement>['type'];
  ghost?: boolean;
  href?: string;
  loading?: boolean;
  onClick?: () => void;
  icon?: ReactChild;
};

type ChildrenProps = {
  icon?: ReactChild;
};

const Children: React.FC<ChildrenProps> = ({ icon, children }) => (
  <>
    {icon && icon}
    {children}
  </>
);

const Button: React.FC<React.PropsWithChildren<ButtonProps>> = ({ children, icon, loading, ...props }) => (
  <BaseButton {...props}>{loading ? <div /> : <Children icon={icon}>{children}</Children>}</BaseButton>
);

export default Button;
