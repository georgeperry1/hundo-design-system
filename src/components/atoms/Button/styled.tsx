import React from 'react';
import styled from 'styled-components';

import { ButtonProps } from '.';

export const BaseButton = styled.button`
  border-radius: 4px;
  background-color: #000;
  color: #fff;
  font-size: 14px;
  height: 48px;
  font-weight: 700;
  outline: none;
  border: 1px solid #000;
  padding: 0 8px;
`;

const Button: React.FC<ButtonProps> = (props) => {
  return <BaseButton {...props} />;
};

export default Button;
