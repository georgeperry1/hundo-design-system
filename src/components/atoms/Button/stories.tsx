import { Story, Meta } from '@storybook/react';

import Button, { ButtonProps } from '.';

export default {
  title: 'Components/Atoms/Button',
  component: Button,
} as Meta;

const Template: Story<ButtonProps> = (args) => <Button {...args}>Button</Button>;

export const Default = Template.bind({});
