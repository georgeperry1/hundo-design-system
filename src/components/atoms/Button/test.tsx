import { render } from '@testing-library/react';

import Button from '.';

describe('Button', () => {
  it('should render', () => {
    const { baseElement } = render(<Button />);

    expect(baseElement).toMatchSnapshot();
  });
});
