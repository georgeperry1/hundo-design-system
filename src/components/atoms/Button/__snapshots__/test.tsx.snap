// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
.c0 {
  border-radius: 4px;
  background-color: #000;
  color: #fff;
  font-size: 14px;
  height: 48px;
  font-weight: 700;
  outline: none;
  border: 1px solid #000;
  padding: 0 8px;
}

<body>
  <div>
    <button
      class="c0"
    />
  </div>
</body>
`;
